### PR TITLE
Convert EcalSelectiveReadoutProducer to stream module

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

The EcalSelectiveReadoutProducer was converted from a one:: to stream:: module as DIGI workflows using large numbers of threads, greater than 64, were losing efficiency because of the module. As part of the conversion, a fillDescriptions method was also added.

To avoid unnecessary coupling of two modules, the static methods in the EDAnalyzer ECalSRCondTools were removed from the module and placed in a namespace.

#### PR validation:

The code compiles and a simple test running the DIGI+HLT steps succeeded.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
@makortel FYI

**Comment by Dr15Jones** [comment]
please test

File: SimCalorimetry/EcalSelectiveReadoutProducers/plugins/EcalSRCondTools.cc [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+      //trigger tower thresholds (from FENIX configuration):
+      const edm::ESHandle<EcalTPGPhysicsConst> hTp = es.getHandle(tpgPhysicsConstToken_);
+      if (!hTp.isValid()) {
+        cout << "EcalTPGPhysicsConst record not found. Check the Cond DB Global tag.\n";
+      } else {
+        const EcalTPGPhysicsConst* tp = hTp.product();
+        const EcalTPGPhysicsConstMap& mymap = tp->getMap();
+        if (mymap.size() != 2) {
+          cout << "Error: TPG physics record is of unexpected size: " << mymap.size()
```
Comment (civanch) [inline SimCalorimetry/EcalSelectiveReadoutProducers/plugins/EcalSRCondTools.cc:129]: @Dr15Jones , is it OK to left cout in this class. Does it mean that it is allowed for "one" producer? My problem with these cout - it is not obvious which class is printing
.

Comment (civanch) [reply SimCalorimetry/EcalSelectiveReadoutProducers/plugins/EcalSRCondTools.cc:129]: #41789  is created , so this PR may be merged.

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**